### DISCUSSION TITLE
fail to launch node icubsrv && icub-head on yarpmanager cluster inside the docker

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#629] [Q&A] [answered]
![2023-05-24 17-44-36 的屏幕截图](https://github.com/robotology/community/assets/7019692/19fb6212-3e6d-44c6-baaa-5d72f5e04750)
The prompt message asked me to type icub@icubsrv's password:  &&   icub@icub-head's password:  after i launch  selected nodes(icubsrv && icub-head) on cluster of yarpmanager, i typed "icub" as password,  but it failed, did I type the wrong password?

While i go back  outside the docker, directly launch yarpserver, yarpmanager and yarpmotorgui on terminal of ubuntu, it works well.

**Comment by pattacini** [comment]
Hi @jimingda 

Simply put, the docker-based system we provide for training is not suitable for being used together with the robot.

Instead, you ought to be using the physical systems connected to the robot made up of the `icubsrv` machines and the robot itself (i.e., `icub-head`).

**Reply by jimingda** [reply]
Thanks, so maybe that is also the reason why I fail to test the training project on icub.

`[MSG] ClusterWidget: yarprun successfully executed on icubsrv
[MSG] ClusterWidget: yarprun successfully executed on icub-head
[MSG] 17 applications are loaded successfully.
[ERR] (Tutorial_on_Joint_Interface_System) cannot run gzclient on localhost : Can't execute command because ?????????
[ERR] (Tutorial_on_Joint_Interface_System) cannot run gzserver on localhost : Can't execute command because ?????????
[ERR] (Tutorial_on_Joint_Interface_System) cannot run gzserver on localhost : Can't execute command because ?????????
[ERR] (Tutorial_on_Joint_Interface_System) cannot run gzclient on localhost : Can't execute command because ?????????`

**Reply by pattacini** [reply]
Yep, definitely.

**Reply by jimingda** [reply]
Thanks ugo, and another question is that if i want implement the tutorial_joint-interface in real icub, which part do i need to configure, maybe change optJoint.put("remote","/icubSim/left_arm"); to optJoint.put("remote","/icub/left_arm") is not enough

**Reply by pattacini** [reply]
Actually, that's sufficient to connect to the real robot.
To this aim, remember to leave aside the docker container and build the tutorial on the robot PC.

**Reply by jimingda** [reply]
Thanks, now there is another problem even i change optJoint.put("remote","/icubSim/left_arm"); to optJoint.put("remote","/icub/left_arm") ;
after i build the tutorial, that the Host of Tutorial_on_Joint_Interface_App and Tutorial_on_Joint_Interface_System shows on YARP module manager is still 'localhost', not the icubsrv or icub-head,  someplace didn't configure well ?

**Reply by pattacini** [reply, marked as answer]
Hi @jimingda 

Probably, you should be learning first the use of [**`yarpmanager`**](https://yarp.it/latest/group__yarpmanager.html).

At any rate, you can always run the executable from the terminal bypassing the manager, which is a thing I suggest you do in the meanwhile.
